test/modules/http/compliance/http-max-header-fields-count.test.ts

```diff
@@ -87,3 +87,40 @@ it('supports multiple parallel "slow" requests', async () => {
     ])
   }
 })
+
+it('supports responses with more than default maximum header fields count', async () => {
+  const responseHeadersPromise = new DeferredPromise<Headers>()
+
+  interceptor.on('request', ({ request, controller }) => {
+    const responseHeadersPairs = Array.from({ length: 60 })
+      .map<[string, string]>((_, index) => {
+        return [`x-response-header-${index}`, index.toString()]
+      })
+      .sort()
+    
+    const response = new Response(null, {
+      status: 200,
+      headers: new Headers(responseHeadersPairs)
+    })
+    
+    controller.respondWith(response)
+  })
+
+  interceptor.on('response', ({ response }) => {
+    responseHeadersPromise.resolve(response.headers)
+  })
+
+  const request = http.request('http://localhost/response-headers')
+  request.end()
+
+  await waitForClientRequest(request)
+  const responseHeaders = await responseHeadersPromise
+
+  const expectedHeadersPairs = Array.from({ length: 60 })
```

Comment:
Let's declare it before the interceptor's request listener, then reuse it in the mocked response and in this assertion. Would that work?